### ISSUE TITLE
No gender input - Female & Male - complete run

### ISSUE THREAD

**Issue by fferrero07** [deeplearningrosario/deeplearningforcomputervision#36]
### Run Specifics

This run's objective was to train a model with the complete female and then male datasets without a gender input. And then compare it with a run that was fed both genders and their flag as input. 

Image layer: ResNet50
Image quantity: 

- Males: 6780 (4520 training - 1130 validation - 1130 testing)
- Females: 5740 (3828 training - 956 validation - 956 testing)

Epochs: 30
Batch size: 32

Loss: Mean Squared Error
Additional metrics: Mean Absolute Error
Optimizer: Adam(lr=0.001, amsgrad=True)
Callbacks: 
- TensorBoard
- Checkpoint
- ReduceLROnPlateau`(monitor="val_loss", factor=0.8, patience=3, verbose=1, min_lr=0.0001)`
- CSVLogger

### Weights
To be found [here](https://drive.google.com/drive/folders/1T_m1SeqoKR8X5-NkS5TdNS_h4Bt0YgkX)

### Statistics
_**Male results:**_

| Test Loss | Test MAE |
| -- | -- |
| 165.45352595506517 | 9.729406171984378 |

![history_loss](https://user-images.githubusercontent.com/10561353/44956648-c8fe2f00-ae9d-11e8-88e4-ebc2ba94b01c.png)

![history_mean](https://user-images.githubusercontent.com/10561353/44956649-ca2f5c00-ae9d-11e8-8a64-1f55cf64b941.png)

**Training:**

Loss   (MSE) |   | MAE |   |
-- | -- | -- | -- 
Range | 3665.87105329395 | Range | 38.9844430906582
Minimum | 70.0073050439885 | Minimum | 6.42861766983977
Maximum | 3735.87835833794 | Maximum | 45.413060760498

_**Female results:**_

| Test Loss | Test MAE |
| -- | -- |
| 252.88233922216185 | 12.213448676105324 |

![history_loss](https://user-images.githubusercontent.com/10561353/44956654-d1ef0080-ae9d-11e8-91b3-c5532be15df8.png)

![history_mean](https://user-images.githubusercontent.com/10561353/44956655-d4515a80-ae9d-11e8-845c-90a7c37c4f8e.png)

**Training:**

Loss   (MSE) |   | MAE |   |
-- | -- | -- | -- 
Range | 3155.21992065365 | Range | 36.8489660245123
Minimum | 70.2708127172264 | Minimum | 6.35141476119947
Maximum | 3225.49073337088 | Maximum | 43.2003807857118

### Male summary table

epoch | loss | mean_absolute_error | val_loss | val_mean_absolute_error | lr
-- | -- | -- | -- | -- | --
0 | 3735.878358 | 45.4130608 | 1906.790736 | 34.65514229 | 0.001
1 | 1475.775394 | 29.9408432 | 1835.54915 | 32.19074962 | 0.001
2 | 852.5461196 | 22.0513992 | 2788.271009 | 38.48999774 | 0.001
3 | 503.5753677 | 16.9660843 | 2768.379967 | 45.57412083 | 0.001
4 | 375.2989975 | 14.9126666 | 461.8376905 | 16.72686494 | 0.001
5 | 327.7298681 | 13.7292592 | 529.7506999 | 17.36135281 | 0.001
6 | 295.1947653 | 13.3938506 | 427.8294012 | 15.2590864 | 0.001
7 | 291.7132728 | 13.1980182 | 755.3697909 | 19.33551435 | 0.001
8 | 285.0570527 | 13.0504262 | 560.0878593 | 19.72545296 | 0.001
9 | 231.3292901 | 11.8644163 | 484.4907453 | 17.36200377 | 0.001
10 | 199.4873125 | 11.0671422 | 644.6251781 | 21.19467672 | 0.0008
11 | 192.5788161 | 10.8343684 | 867.3276526 | 25.41808887 | 0.0008
12 | 213.7923489 | 11.4480356 | 904.1453503 | 22.94275237 | 0.0008
13 | 193.6848752 | 10.8102066 | 416.5463125 | 16.44532143 | 0.00064
14 | 161.5025711 | 9.86416686 | 506.4735631 | 18.47999011 | 0.00064
15 | 162.2640958 | 9.88499958 | 307.6508288 | 13.04809371 | 0.00064
16 | 128.5796792 | 8.82230294 | 434.3318476 | 16.97257988 | 0.00064
17 | 132.8018082 | 9.02053354 | 253.7780787 | 11.73466675 | 0.00064
18 | 132.5447125 | 8.98652808 | 389.353908 | 14.13310813 | 0.00064
19 | 120.0633321 | 8.5653122 | 289.865596 | 12.30641432 | 0.00064
20 | 138.5781164 | 9.14525234 | 514.0705508 | 15.80191646 | 0.00064
21 | 133.5123427 | 8.88643699 | 2831.086421 | 43.40338214 | 0.000512
22 | 127.4677474 | 8.66647809 | 280.451848 | 12.66906182 | 0.000512
23 | 101.6274139 | 7.75345884 | 306.8332981 | 13.33895176 | 0.000512
24 | 94.85117663 | 7.51374074 | 210.201033 | 10.66568768 | 0.0004096
25 | 99.02621717 | 7.68122114 | 294.5820907 | 12.66259017 | 0.0004096
26 | 86.85516433 | 7.24124205 | 281.7136742 | 11.84871616 | 0.0004096
27 | 86.90139014 | 7.23321914 | 213.4823761 | 10.27092539 | 0.0004096
28 | 81.33587895 | 6.93781652 | 230.3643531 | 11.31688888 | 0.00032768
29 | 70.00730504 | 6.42861767 | 171.0915767 | 9.570525293 | 0.00032768

### Female summary table

epoch | loss | mean_absolute_error | val_loss | val_mean_absolute_error | lr
-- | -- | -- | -- | -- | --
0 | 3225.490733 | 43.20038079 | 1576.306879 | 31.22566651 | 0.001
1 | 1076.844283 | 25.9459019 | 2222.802215 | 37.31496889 | 0.001
2 | 651.8678846 | 19.70350236 | 2870.83015 | 44.97444483 | 0.001
3 | 424.2589085 | 16.029845 | 479.0427026 | 16.96131907 | 0.001
4 | 468.9475468 | 16.86651632 | 3842.27944 | 52.12726639 | 0.001
5 | 347.3564427 | 14.31771837 | 973.4068869 | 24.36733318 | 0.001
6 | 283.2564641 | 13.13759761 | 479.085034 | 17.24283487 | 0.001
7 | 297.662522 | 13.34673126 | 392.771664 | 14.7479914 | 0.0008
8 | 236.5183763 | 11.90810465 | 437.3649416 | 15.08474613 | 0.0008
9 | 213.2091964 | 11.30913243 | 1337.481199 | 30.01092569 | 0.0008
10 | 199.4322018 | 11.0171653 | 595.9390637 | 18.9323482 | 0.0008
11 | 175.9386262 | 10.34814347 | 212.6212804 | 10.9363087 | 0.00064
12 | 167.2587407 | 10.01780319 | 412.8584431 | 15.51453798 | 0.00064
13 | 160.5880615 | 9.870435453 | 410.5734939 | 15.32451994 | 0.00064
14 | 160.2405254 | 9.871293189 | 472.8765611 | 16.81057597 | 0.00064
15 | 159.4876601 | 9.627847321 | 297.5227997 | 12.3094833 | 0.000512
16 | 138.8647305 | 9.159289393 | 254.3629628 | 11.98408685 | 0.000512
17 | 139.8174202 | 9.207065934 | 235.3499366 | 11.5440521 | 0.000512
18 | 115.7415817 | 8.279428336 | 203.3476367 | 10.96318661 | 0.0004096
19 | 105.3902059 | 8.009345714 | 205.9090334 | 10.92592321 | 0.0004096
20 | 101.2961557 | 7.819296212 | 189.6300927 | 10.34817419 | 0.0004096
21 | 109.2503866 | 8.160289246 | 283.7074635 | 12.97696499 | 0.0004096
22 | 90.35348151 | 7.319554613 | 315.9518601 | 13.60090854 | 0.0004096
23 | 88.33994647 | 7.246747578 | 257.1933936 | 12.33918425 | 0.0004096
24 | 83.56512654 | 6.937798002 | 263.8342201 | 12.30039209 | 0.00032768
25 | 94.79786806 | 7.462437035 | 222.1719248 | 11.17091251 | 0.00032768
26 | 81.94399527 | 6.891420847 | 249.7709405 | 12.07628296 | 0.00032768
27 | 87.12744164 | 7.165717904 | 216.6003566 | 11.32714567 | 0.000262144
28 | 74.93815445 | 6.647280035 | 222.8052986 | 11.53205572 | 0.000262144
29 | 70.27081272 | 6.351414761 | 282.6419747 | 13.16427019 | 0.000262144

### Run Logs did not fit due to max body length

[male output 20180628.docx](https://github.com/deeplearningrosario/deeplearningforcomputervision/files/2343416/male.output.20180628.docx)

[female output 20180625.docx](https://github.com/deeplearningrosario/deeplearningforcomputervision/files/2343417/female.output.20180625.docx)
